### PR TITLE
Styling the app - Milestone 5

### PR DESCRIPTION
**Pull request by MahmoudBakr23**
### Hello reviewer!

**In this milestone I did the following:**

- Style the app's components according to the given design template.

>Please note that I cannot push to Heroku until I merge this branch to the master 😄

<hr>

_Please let me know if there is any requirement I missed._ 🙏
_Thank you for your hard work._ ❤️

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,266 @@
...
+  height: 100%;
+  background-color: #fff;
+  padding: 1rem 5rem;
+}
+
+nav {
+  display: flex;
+  justify-content: space-between;
+  align-content: center;
+  align-items: center;
```
Comment (rindrajosia) [inline src/App.css]: - [ ] [REQUIRED] In the stylesheets of the app, make sure that basic CSS bad practices are not used ([article for support ](https://speckyboy.com/good-bad-css-practices)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright)

no repeating of the same rule
Please correct those issues and send another review request, good luck, and well done so far!

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,266 @@
...
+  padding: 3rem 5rem;
+  display: grid;
+  grid-template-columns: repeat(1, 1fr);
+}
+
+.book-container {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  align-content: center;
```
Comment (rindrajosia) [inline src/App.css]: no repeating of the same rule

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [ ] Styling should be **EXACTLY** the same as the one provided in the designs ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone5/exactlySameAsDesign.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/45791599/))
The ADD Book button becomes very large when I zoom in on your app.


- [ ] Please, use tools like Grammarly to help you with your readme and don't forget to update your live demo. Keep in mind that your readme file is the first thing users check.



_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/App.css (hunk 2 above)
Comment (MahmoudBakr23) [reply src/App.css]: **Thank you for your review** 😄

- The element is stretching to take the full width of the screen when you zoom and the screen size gets changed! In the normal 100% screen size, I think my design is ok!
- There's not any repetition! they're `align-items` and `align-content`!!
- Also, I checked my README file text and Grammarly doesn't show any errors! 😄

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @MahmoudBakr23, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- [ ] Kindly update the live version to match the local one. 
- [ ] Kindly fix the repeated CSS rules as highlighted in the previous review. For example `.nav-items line 46` `nav line 23` and `.book-container line 99-102` have the same rules and they can be grouped together.
- [ ] Great job on the design. However for it to match the given design, Kindly remove the Select Categories and move the categories in the navbar `categories` section next to `Books`

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by MahmoudBakr23** [comment]
**Thanks for your review** 😄 

- Kindly note that I can't push to Heroku master unless I merge all the changes to the master branch, and I can't do that before I get the approval! Once I get it, I'll update the live demo link! + It's not required in this milestone, and deployment has a separate step that I have to complete in this module. 😄 🙏 
- Please put in mind that the design is delivered exactly like the given one, and the categories filter is an additional element that I see fits the design as it is, can you please tell me why should I change it? + If I moved the `Filter` component to the `Navbar` component, then I'll have to move the functionality of changing the category to the Navbar component which isn't what was required in the previous milestone and it'll be inconsistent with the design. 😄

**Review by george-shammar — APPROVED** [review]
Hi @MahmoudBakr23 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by george-shammar** [comment]
> **Thanks for your review** smile
> 
>     * Kindly note that I can't push to Heroku master unless I merge all the changes to the master branch, and I can't do that before I get the approval! Once I get it, I'll update the live demo link! + It's not required in this milestone, and deployment has a separate step that I have to complete in this module. smile pray
> 
>     * Please put in mind that the design is delivered exactly like the given one, and the categories filter is an additional element that I see fits the design as it is, can you please tell me why should I change it? + If I moved the `Filter` component to the `Navbar` component, then I'll have to move the functionality of changing the category to the Navbar component which isn't what was required in the previous milestone and it'll be inconsistent with the design. smile

- You can definately update an heroku app if you have not deployed it through the master in the first instance. However, since it's not a mandatory requirement, your project is approved. You can now update the app. Congratulations and good job.

**Pull request merged**